src/user_input_compiler.rs

```diff
@@ -0,0 +1,258 @@
+//! # user_input_compiler
+//!
+//! Parses Input with this Grammer after trimming the Input and ignoring Whitespaces:
```

Comment:
Small detail: "Grammer" should be "grammar" and "Input" should be "input", see https://www.youtube.com/watch?v=N4vf8N6GpdM :grin: